app/src/test/java/waiter/CommunicatorTest.java

```diff
@@ -1,28 +1,33 @@
 package waiter;
 
-import waiter.ClientConnection.Connectable;
 import waiter.Communicator.Communicator;
 import waiter.Listener.mock.ListenerMock;
 import waiter.Messenger.mock.MessengerMock;
-
-import org.junit.jupiter.api.Test;
 import waiter.Reactor.mock.ReactorMock;
 
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
 import java.io.IOException;
 
 class CommunicatorTest {
 
+    String clientHasDisconnected;
+
+    @BeforeEach
+    void setUp() {
+        clientHasDisconnected = null;
+    }
+
     @Test
     void communicatesProvidedInput() throws IOException {
-        ListenerMock listenerMock = new ListenerMock("foo");
+        String[] clientRequests = {"curl foo1", "curl foo2", "curl foo3", "curl foo4"};
+        ListenerMock listenerMock = new ListenerMock(new String[]{"foo", clientHasDisconnected});
         MessengerMock messengerMock = new MessengerMock();
         Communicator communicator = new Communicator(listenerMock, messengerMock);
-        ReactorMock reactor = new ReactorMock(5);
+        ReactorMock reactor = new ReactorMock(clientRequests);
         communicator.communicate(reactor);
-        Connectable clientConnectionMock = messengerMock.calledWith;
-        assertEquals("foo", clientConnectionMock.read().open());
-        assertEquals(reactor.currentNumberOfEchos, 5);
+        assertEquals(clientRequests.length, reactor.numberOfAcceptedClients);
```

Comment:
Its not so much client input as it is client requests to connect to the server, so it is trying to say that 4 separate connections (via 4 curl -v telnet://localhost:4424) were sent one after another (one connects, echoes a message and then disconnects, then the next connection is made and so on). It is done for times and then the server is set to close, in which case it exits the for loop, I hope this makes sense!

Reply:
Make sense. I think the confusion is coming from the name of this test which is `communicatesProvidedInput`. This test is more like `shouldAllowReconnect` or something like that if I'm understanding correctly?

Reply:
That is a good point, I think the `communicatesProvidedInput` test name is outdated, `shouldAllowReconnect` makes more sense but I think is still a little misleading as it can allow for both reconnection and another connection if that makes sense, so maybe `shouldAllowSequentialConnections` or something, if that reasoning makes sense 😄

Reply:
Addressed in [this commit](https://github.com/jonmanahan/waiter/pull/7/commits/97d783bb1cde52ca12d9dc231d05bf967e161ca5), fixed the test name, some more test format and added a comment to the null variable declaration.